modules/local-notifications/src/main/java/com/gluonhq/attach/localnotifications/impl/IOSLocalNotificationsService.java

```diff
@@ -52,10 +72,42 @@ protected void unscheduleNotification(Notification notification) {
     
     @Override
     protected void scheduleNotification(Notification notification) {
+        createImageInAssets(notification.getImageInputStream());
         registerNotification(notification.getTitle() == null ? "" : notification.getTitle(), 
                 notification.getText(), notification.getId(), notification.getDateTime().toEpochSecond());
     }
-    
+
+    private void createImageInAssets(InputStream imageInputStream) {
+        File file = new File(assetsFolder, NOTIFICATION_IMAGE);
+        if (file.exists()) {
+            file.delete();
+        }
+
+        try (imageInputStream) {
+            copyFile(imageInputStream, file.getAbsolutePath());
+        } catch (IOException ex) {
+            LOGGER.log(Level.WARNING, "Error reading image file", ex);
```

Comment:
Shouldn't the caller somehow be informed that copying failed?

Reply:
If the copying failed, there is already an exception logged. Probably not critical enough to interrupt the whole notification process?
Or do we need to foward the exception up to the API level?